.changeset/large-olives-press.md

```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fix `declaration-block-no-redundant-longhand-properties` autofix conflicts
```

Comment:
[suggest] Let's clarify the resolve bug:

```suggestion
Fix `declaration-block-no-redundant-longhand-properties` autofix conflicts for `border-*`
```

Reply:
See https://github.com/stylelint/stylelint/pull/7626#issuecomment-2065876959. The following may be more accurate:

```suggestion
Fix `declaration-block-no-redundant-longhand-properties` false positives for `border-*-style`
```

Reply:
This is the only example that I could find.
The underlying question is, if we care about perf, should we put an `if` around the `forEach`?
i.e. `if (blacklist.includes(foo)) {…}`

Reply:
> for `border-*`

It's inaccurate. Only a subset of `border-*` are affected and the current implementation doesn't differentiate.
i.e. it's future proof but probably could use a blacklist instead

Reply:
I have added the blacklist in the last commit.